backend/src/apiserver/server/experiment_server.go

```diff
@@ -39,19 +38,7 @@ func (s *ExperimentServer) GetExperiment(ctx context.Context, request *api.GetEx
 
 func (s *ExperimentServer) ListExperiment(ctx context.Context, request *api.ListExperimentsRequest) (
 	*api.ListExperimentsResponse, error) {
-	var opts *list.Options
-	var err error
-	if request.PageToken != "" {
-		opts, err = list.NewOptionsFromToken(request.PageToken, int(request.PageSize))
-	} else {
-		var f *api.Filter
-		f, err = parseAPIFilter(request.Filter)
-		if err != nil {
-			return nil, err
-		}
-
-		opts, err = list.NewOptions(&model.Experiment{}, int(request.PageSize), request.SortBy, f)
-	}
+	opts, err := validatedListOptions(&model.Experiment{}, request.PageToken, int(request.PageSize), request.SortBy, request.Filter)
```

Comment:
nit consider making &model.Experiment{} a constant so it doesn't instantiate a new object for each API call.

Reply:
Hmm, I think any performance issue here is negligible, and the benefits of having this right here for readability (vs another indirection for the reader) outweighs it anyway. It's really cheap to create it and the performance cost of this call is elsewhere, so I'd rather leave it as is. What do you think?

Reply:
sgtm